### ISSUE TITLE
Option to add existing files to a sketch

### ISSUE THREAD

**Issue by Lorandil** [arduino/arduino-ide#284]
### Describe the request

It would be great if the files drop down menu would allow to "add existing files"

### Describe the current behavior

If I want to reuse some code I wrote earlier without creating a library, I manually copy those files to the sketch folder.

~~At the moment I have to restart the IDE to integrate them into the sketch.~~ (tracked at https://github.com/arduino/arduino-ide/issues/296)

### Expected behavior

It is possible to add existing code files to a sketch via the Arduino IDE GUI.

### Arduino IDE version

76f9f63

### Operating system

Windows

### Operating system version

10

### Additional context

Additional reports:

- https://forum.arduino.cc/t/sketch-add-file/677557
- https://forum.arduino.cc/t/adding-files-change-of-behavior-bug-or-me-being-dumb/1068993

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://www.arduino.cc/en/software#nightly-builds)
- [X] My report contains all necessary details

**Comment by ubidefeo** [comment]
@per1234 
This one is something I have also found myself in need of.
What's your opinion?

**Comment by per1234** [comment]
I don't think it would do any harm. This tabs menu is not something a beginner needs to use, so adding additional functionality to it doesn't add complexity to the beginner's UI.

**Comment by ubidefeo** [comment]
thanks @per1234 

let's create a task for this :)

**Comment by per1234** [comment]
Thanks to [a forum user](https://forum.arduino.cc/t/adding-files-change-of-behavior-bug-or-me-being-dumb/1068993), I just discovered that Arduino IDE 1.x has this capability, so this is actually a matter of feature parity.

I remember that when the **Sketch > Add File...** capability was added to Arduino IDE 2.x, it was specified as a way to add supplemental files (e.g., documentation) to the `data` subfolder of the sketch. I never used this feature in Arduino IDE 1.x, but now that I try it out, I see the Arduino IDE 1.x behavior is more complex than what was specified for the Arduino IDE 2.x implementation:

- If the selected file has a supported sketch code file extension (e.g., `.ino`, `.h`, `.cpp`, `.c`, `.S`), add it to the the root of the sketch folder.
- If the selected file has any other file extension, add it to the `data` subfolder of the sketch.

Arduino IDE 2.x's **Sketch > Add File...** capability always does the second of these, even when adding a file with a supported sketch code file extension.
